### PR TITLE
Fix some fuzzer issues

### PR DESCRIPTION
**Pull request by hannes**
- https://github.com/duckdb/duckdb-fuzzer/issues/2688
- https://github.com/duckdb/duckdb-fuzzer/issues/2692

### CODE REVIEW

**Comment by hannes** [comment]
@taniabogatsch can you maybe have a look at the array_slice changes? Seems like a rather complex function

File: src/core_functions/scalar/list/array_slice.cpp [hunk 1]
```diff
@@ -163,6 +163,18 @@ static void ExecuteConstantSlice(Vector &result, Vector &str_vector, Vector &beg
                                  optional_ptr<Vector> step_vector, const idx_t count, SelectionVector &sel,
                                  idx_t &sel_idx, optional_ptr<Vector> result_child_vector, bool begin_is_empty,
                                  bool end_is_empty) {
+
+	// check all this nullness early
```
Comment (taniabogatsch) [inline src/core_functions/scalar/list/array_slice.cpp:167]: This early-out looks good to me! `sel_valid` can only ever be `true` if none of this nullness applies. All other cases are also never reached if any of these is not valid.

File: src/core_functions/scalar/list/array_slice.cpp [hunk 2]
```diff
@@ -180,28 +192,23 @@ static void ExecuteConstantSlice(Vector &result, Vector &str_vector, Vector &beg
...
 
-	auto str_valid = !ConstantVector::IsNull(str_vector);
-	auto begin_valid = !ConstantVector::IsNull(begin_vector);
-	auto end_valid = !ConstantVector::IsNull(end_vector);
-	auto step_valid = step_vector && !ConstantVector::IsNull(*step_vector);
-
 	// Clamp offsets
 	bool clamp_result = false;
-	if (str_valid && begin_valid && end_valid && (step_valid || step == 1)) {
+	if ((step_valid || step == 1)) {
```
Comment (taniabogatsch) [inline src/core_functions/scalar/list/array_slice.cpp:197]: Double-brackets nit.

**Review by taniabogatsch — APPROVED** [review]

File: src/core_functions/scalar/list/array_slice.cpp (hunk 2 above)
Comment (hannes) [reply src/core_functions/scalar/list/array_slice.cpp]: Thanks fixed

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**